InfluxDB: Adds start page for logs in Explore

kaydelaney opened the pull request:
**What this PR does / why we need it**:
Adds a start page / cheat sheet giving instructions on what to do to get started when the `Logs` mode is selected and there are no measurements or fields selected.
Needed to guide the user in how to visualize logs.

**Which issue(s) this PR fixes**:
Closes #17499

**Special notes for your reviewer**:

kaydelaney committed:
Influx: Adds start page for logs in Explore
Closes #17499

davkal commented on public/app/plugins/datasource/influxdb/components/InfluxCheatSheet.tsx:
```diff
@@ -0,0 +1,20 @@
+const CHEAT_SHEET_ITEMS = [
+  {
+    title: 'Getting started',
+    label: 'Start by selecting a measurement and field from the dropdown above.',
```
... and then hit "Run Query".

Also would be good to describe the tag chooser to narrow down your search.

kaydelaney replied:
👍

davkal commented on public/app/features/explore/Explore.tsx:
```diff
@@ -268,7 +268,7 @@ export class Explore extends React.PureComponent<ExploreProps> {
                   <main className="m-t-2" style={{ width }}>
                     <ErrorBoundary>
-                      {showingStartPage && <StartPage onClickExample={this.onClickExample} />}
+                      {showingStartPage && <StartPage onClickExample={this.onClickExample} mode={mode} />}
```
The page is not shown after switching from Logs to Metrics and then back to Logs.

kaydelaney replied:
Good catch! Will check this out

marefr replied:
I just merged my PR for clearing queries when switching between metrics and logs (#17505) so please merge with master. I think that should fix this.

davkal commented on packages/grafana-ui/src/types/datasource.ts:
```diff
@@ -281,6 +281,7 @@ export interface ExploreQueryFieldProps<
 
 export interface ExploreStartPageProps {
   onClickExample: (query: DataQuery) => void;
+  mode?: any;
```
This could use a comment on what `mode` could be used for.

marefr replied:
Think mode property can be removed for now based on https://github.com/grafana/grafana/pull/17521#issuecomment-500870094

marefr commented:
Added a comment in the referenced issue. TLDR; think we should merge this showing startpage in both metrics and logs mode for now. We can make this work better later.

kaydelaney committed:
Influx: Rebased with master and removed conditional display of cheat sheet in explore
Cheat sheet now displays during both Metrics and Logs mode, though this is only
temporary.

marefr reviewed (approved):
LGTM
